cmake/Modules/FindPAPI.cmake

```diff
@@ -70,25 +71,28 @@ if(PAPI_INCLUDE_DIR)
         # find the components
         enable_language(C)
         foreach(component IN LISTS PAPI_FIND_COMPONENTS)
-            file(WRITE "${PROJECT_BINARY_DIR}/papi_${component}_detect.c"
+            set(SRC_FILE "${WORK_DIR}/papi_${component}_detect.c")
+            set(BIN_FILE "${WORK_DIR}/papi_${component}_detect.bin")
+            file(WRITE "${SRC_FILE}"
                 "
                 #include <papi.h>
                 int main() {
                  int retval;
                  retval = PAPI_library_init(PAPI_VER_CURRENT);
                    if (retval != PAPI_VER_CURRENT && retval > 0)
                     return -1;
-                   if (PAPI_get_component_index(\"${component}\") < 0)
+                   if (PAPI_get_component_index(\"${component}\") == PAPI_ENOCMP)
                     return 0;
                    return 1;
                 }"
                 )
             try_run(PAPI_${component}_FOUND
                 gko_result_unused
-                "${PROJECT_BINARY_DIR}"
-                "${PROJECT_BINARY_DIR}/papi_${component}_detect.c"
+                "${WORK_DIR}"
+                "${SRC_FILE}"
                 CMAKE_FLAGS -DINCLUDE_DIRECTORIES=${PAPI_INCLUDE_DIR}
                 LINK_LIBRARIES ${PAPI_LIBRARY}
+                COPY_FILE ${BIN_FILE}
```

Comment:
is the copy_file only for the debug?

Reply:
Yes, then you can run the produced binary manually to debug it. I realize this is what, e.g., OpenMP does, so I'm doing the same here.